src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1550,6 +1550,14 @@ protected override void ProcessRecord()
                                     OutFile = null;
                                 }
 
+                                // Detect insecure redirection
+                                if (response.RequestMessage.RequestUri.Scheme == "https" && response.Headers.Location?.Scheme == "http")
+                                {
+                                        ErrorRecord er = new(new InvalidOperationException(), "InsecureRedirection", ErrorCategory.InvalidOperation, request);
+                                        er.ErrorDetails = new ErrorDetails(WebCmdletStrings.InsecureRedirection);
+                                        WriteError(er);
```

Comment:
Shouldn't this be a terminating error here?  There doesn't seem to be any point in continuing and we should throw `ThrowTerminatingError()` here.

Reply:
I used `WriteError()` to mirror the behaviour of lines 1604-1609:

```cs
if (WebSession.MaximumRedirection == 0 && IsRedirectCode(response.StatusCode)) // Indicate "HttpClientHandler.AllowAutoRedirect == false"
{
    ErrorRecord er = new(new InvalidOperationException(), "MaximumRedirectExceeded", ErrorCategory.InvalidOperation, request);
    er.ErrorDetails = new ErrorDetails(WebCmdletStrings.MaximumRedirectionCountExceeded);
    WriteError(er);
}
```
I can change it if needed.

Reply:
What is a behavior of other utils in the scenario?

Reply:
Also text in WebCmdletStrings.InsecureRedirection assumes we should throw.

Reply:
Test:
`Invoke-WebRequest "https://mockbin.org/redirect/308?to=https://mockbin.org/redirect/302?to=http://mockbin.org/status/200" -SkipHttpErrorCheck` --> 302
`curl -L "https://mockbin.org/redirect/308?to=https://mockbin.org/redirect/302?to=http://mockbin.org/status/200"` --> 200

(HttpClient does not seem to allow insecure redirect: https://github.com/dotnet/runtime/issues/23801)

Reply:
Please add new test.